_pytest/test_util.py

```diff
@@ -0,0 +1,43 @@
+from rasa_nlu import util
+from rasa_nlu import config_keys
+
+
+def test_update_config():
```

Comment:
As you mentioned in the issue, this is related to explain the overriding order. Unfortunately, I find that hard to understand from the tests.

I'd prefer a single test for each case with a descriptive name ala `test_config_overrides_env`, ... The code wouldn't be any longer and the test name would provide a description of the specific test case.